server/models/DeleteAccountRequest.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  AllowNull,
+  Column,
+  CreatedAt,
+  DataType,
+  ForeignKey,
+  Model,
+  Table,
+  UpdatedAt,
+} from "sequelize-typescript";
+import { User } from "./User";
+import { DeleteAccountRequestStatus } from "@server/types/deleteaccountrequests";
+
+@Table({
+  timestamps: true,
+  tableName: "delete_account_requests",
+})
+export class DeleteAccountRequest extends Model {
+  @ForeignKey(() => User)
+  @AllowNull(false)
+  @Column(DataType.STRING)
+  declare user_id: string;
+
+  @Column(DataType.ENUM("pending", "completed"))
+  declare status: DeleteAccountRequestStatus;
+
+  @Column(DataType.DATE)
+  declare requested_at: Date;
+
+  @Column(DataType.DATE)
```

Comment:
If we are physically removing the row in `users` on delete, this record will be cascade deleted too. Not sure if there's a need for `completed_at`.